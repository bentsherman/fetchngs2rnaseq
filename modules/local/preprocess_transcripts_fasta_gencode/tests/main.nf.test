nextflow_process {

    name "Test Process PREPROCESS_TRANSCRIPTS_FASTA_GENCODE"
    script "../main.nf"
    process "PREPROCESS_TRANSCRIPTS_FASTA_GENCODE"

    test("gencode fasta") {

        when {
            process {
                """
                def fileContents = [
                    '>chr22|removeme',
                    'ACTCAAGATAATGATGAGTAAAGAATATATTTCTAACAACAAAAAGGAAATTTGATAGTA'
                ]
                def gencodeFasta = file("${workDir}/gencodeFasta.fasta")
                gencodeFasta.withWriter{ out ->
                    fileContents.each {out.println it}
                }
                input[0] = gencodeFasta
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
