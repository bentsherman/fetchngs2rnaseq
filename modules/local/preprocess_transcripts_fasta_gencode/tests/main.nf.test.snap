{
    "gencode fasta": {
        "content": [
            {
                "0": [
                    "gencodeFasta.fixed.fa:md5,e9267fa40efc784cc92575b5ba8a441b"
                ],
                "1": [
                    "versions.yml:md5,e2c7f636c472887a914b128d1444737e"
                ],
                "fasta": [
                    "gencodeFasta.fixed.fa:md5,e9267fa40efc784cc92575b5ba8a441b"
                ],
                "versions": [
                    "versions.yml:md5,e2c7f636c472887a914b128d1444737e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-06T16:16:36.494744"
    }
}